src/openforms/api/serializers.py

```diff
@@ -85,15 +112,79 @@ def get_child_serializer_class(self):
     def process_object(self, obj):
         return obj
 
-    def create(self, validated_data):
-        model = self.get_child_serializer_class().Meta.model
-
-        objects_to_create = []
-        for data_dict in validated_data:
-            obj = model(**data_dict)
-            objects_to_create.append(self.process_object(obj))
+    def _batch(
+        self,
+        data: Iterable[MutableMapping[str, Any]],
+    ) -> Iterable[BatchItem]:
+        # Helps create() and validate() deal with nested serializers in O(1) db
+        # queries. i.e. one insert per model type.
+        #
+        # It helps by yielding BatchItems of unsafed Model instances. Nested
+        # models should be safed first, so parent gets saved with the new
+        # foreign keys. The order of BatchItems is defined such that high
+        # priority comes first, and models of the same type are grouped
+        # together for a single bulk_create.
+
+        child_serializer = self.get_child_serializer_class()
+        model = child_serializer.Meta.model
+        child_fields = child_serializer().get_fields()
+        prio = 0
+
+        for index, data_dict in enumerate(data):
+            for field_name, sub_data in data_dict.items():
+                if field_name not in child_fields:  # pragma: no cover
+                    continue
+                source = f"{index}.{field_name}"
+                field = child_fields[field_name]
+                if isinstance(field, serializers.ModelSerializer):
+                    # Turn sub_data into *unsaved* instance
+                    # if only drf Serializers had a "build" method that just
+                    # instantiated without saving
+                    sub_model = field.Meta.model
+                    sub_instance = (
+                        sub_data
+                        if isinstance(sub_data, sub_model)
+                        else sub_model(**sub_data)
```

Comment:
DRF turns the data into instances on the validate pass, so in the create pass, no sub_model call is needed anymore.